### PR TITLE
Move ntf_queue to proper NotificationQueue class

### PR DESCRIPTION
**Pull request by kcudnik**

### CODE REVIEW

**Comment by kcudnik** [comment]
build fail caused by https://github.com/Azure/sonic-swss-common/pull/318 and missing configuration in build docker

File: syncd/NotificationQueue.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include "syncd.h"
```
Comment (qiluo-msft) [inline syncd/NotificationQueue.h]: Can you remove this `include`?

Comment (kcudnik) [reply syncd/NotificationQueue.h]: it's actually required since includes sai.h and defines _In_ macros used here

**Comment by kcudnik** [comment]
retest this please

**Comment by kcudnik** [comment]
interesting, i have the same aspell installed and my didn't complained at that word:
Running Aspell
Words to check: 2221
International Ispell Version 3.1.20 (but really Aspell 0.60.7)
 KeyOpFieldsValuesTuple 1547
 UpdateObjectStatus 3148
 WRED 9 3288: RED, WIRED, WEED, WED, FRED, BRED, CRED, WREN, WE'D

File: syncd/NotificationQueue.h (hunk 1 above)
Comment (kcudnik) [reply syncd/NotificationQueue.h]: i removed this include in https://github.com/Azure/sonic-sairedis/pull/541, introducing sal.h and adding it to swss-common with related PR

Comment (qiluo-msft) [reply syncd/NotificationQueue.h]: If you need sai.h, just include the smaller one.

---
In reply to: [352917947](https://github.com/Azure/sonic-sairedis/pull/539#discussion_r352917947) [](ancestors = 352917947)

Comment (kcudnik) [reply syncd/NotificationQueue.h]: done

**Comment by kcudnik** [comment]
test this please

**Pull request merged**